app/controllers/pages/conditions_controller.rb

```diff
@@ -54,13 +54,30 @@ def delete
     render template: "pages/conditions/delete", locals: { condition_form: }
   end
 
+  def destroy
+    condition = Condition.find(params[:condition_id], params: { form_id: @form.id, page_id: page.id })
+
+    case condition_form_params[:confirm_deletion]
+    when "false"
+      redirect_to edit_condition_path(@form.id, page.id, params[:condition_id])
+    when "true"
+      if condition.destroy
+        redirect_to form_pages_path(@form.id, page.id)
+      else
+        render template: "pages/conditions/delete", locals: { condition_form: }
```

Comment:
Shouldn't it be an error if the condition can't be destroyed?

Reply:
I've added the 422 error, I think that'd be the appropriate one here

Reply:
Good point! I've also spotted another test set that I should really build out

Reply:
It feels like something that shouldn't happen in normal operation, so not sure if it needs a test. I think it would only happen if there was something in a callback going on, or a network error? But we should at least log to sentry...

Reply:
I've added an error that's raised when the deletion doesn't take place, as well as a test for the destroy controller action.

Reply:
Looks good!